lib/cfa/base_model.rb

```diff
@@ -53,14 +53,15 @@ def load
     # smart to at first modify existing value, then replace commented out code
     # and if even that doesn't work, then append it at the end
     # @note prefer to use specialized methods of children
-    def generic_set(key, value)
-      modify(key, value) || uncomment(key, value) || add_new(key, value)
+    def generic_set(key, value, tree = data)
+      modify(key, value, tree) || uncomment(key, value, tree) ||
+        add_new(key, value, tree)
     end
 
     # powerfull method that gets unformatted any value in config.
     # @note prefer to use specialized methods of children
-    def generic_get(key)
-      data[key]
+    def generic_get(key, tree = data)
```

Comment:
Using `data` works there? AFAIK, `tree` will be always initialized to the value that `data` (if defined) had when the method was defined.